### ISSUE TITLE
Errors when trying to build iOS project

### ISSUE THREAD

**Issue by magnuswikhog** [florent37/Flutter-AssetsAudioPlayer#33]
On Android it works fine, but I can't build my project in XCode due to the following errors:

assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:114:65: 'AVAudioSessionCategoryPlayback' has been renamed to 'AVAudioSession.Category.playback'

assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:114:65: 'AVAudioSessionCategoryPlayback' was obsoleted in Swift 3 (AVFoundation.AVAudioSessionCategoryPlayback)

assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:114:103: 'AVAudioSessionModeDefault' has been renamed to 'AVAudioSession.Mode.default'

assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:114:103: 'AVAudioSessionModeDefault' was obsoleted in Swift 3 (AVFoundation.AVAudioSessionModeDefault)

assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:116:36: Ambiguous reference to member 'setCategory'

**Comment by MarkinDionisio** [comment]
The same thing happened to me

**Comment by MarkinDionisio** [comment]
I changed the version swift to 4.0 and voilá !

> On Android it works fine, but I can't build my project in XCode due to the following errors:
> 
> assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:114:65: 'AVAudioSessionCategoryPlayback' has been renamed to 'AVAudioSession.Category.playback'
> 
> assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:114:65: 'AVAudioSessionCategoryPlayback' was obsoleted in Swift 3 (AVFoundation.AVAudioSessionCategoryPlayback)
> 
> assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:114:103: 'AVAudioSessionModeDefault' has been renamed to 'AVAudioSession.Mode.default'
> 
> assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:114:103: 'AVAudioSessionModeDefault' was obsoleted in Swift 3 (AVFoundation.AVAudioSessionModeDefault)
> 
> assets_audio_player-1.2.3/ios/Classes/SwiftAssetsAudioPlayerPlugin.swift:116:36: Ambiguous reference to member 'setCategory'

**Comment by dannycortesv** [comment]
@MarkinDionisio How did you changed swift to 4.0?

**Comment by MarkinDionisio** [comment]
> @MarkinDionisio How did you changed swift to 4.0?

Open up your project file in Xcode and check the setting under Build Settings -> Deployment, as shown in the screenshot below:

<img width="765" alt="intro deployment" src="https://user-images.githubusercontent.com/34717943/76445769-b32bcd00-63a4-11ea-83f1-d4df01640c00.png">

**Comment by tonypottera24** [comment]
I got some error messages like this
`note: 'AVCapturePhotoSettings' has been marked as being introduced in iOS 10.0 here, but the deployment target is iOS 8.0.0`

**Comment by geekydiamond** [comment]
Made this change to SwiftAssetsAudioPlayerPlugin.swift line 113
```
           if #available(iOS 10.0, *) {
                try AVAudioSession.sharedInstance().setCategory(AVAudioSession.Category.playback, mode: AVAudioSession.Mode.default, options: [])
            } else {
                try AVAudioSession.sharedInstance().setCategory(AVAudioSession.Category.playback, options: AVAudioSession.CategoryOptions.mixWithOthers)
            }
```
Seems to have fixed the build errors and first tests show audio playing successfully...

**Comment by florent37** [comment]
for ios please follow this instruction https://github.com/florent37/Flutter-AssetsAudioPlayer/issues/29#issuecomment-590537548
